ccl/multiregionccl: TestRegionAddDropWithConcurrentBackupOps failed

ccl/multiregionccl.TestRegionAddDropWithConcurrentBackupOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3253295&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3253295&tab=artifacts#/) on master @ [4e79977415d3101726a65b47f2b084d4a36931b8](https://github.com/cockroachdb/cockroach/commits/4e79977415d3101726a65b47f2b084d4a36931b8):


```
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:711
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:1320
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:289
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:457
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:271
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:174
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils.TestingCreateMultiRegionCluster
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go:78
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:711
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | testing.tRunner
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/usr/local/go/src/testing/testing.go:1193
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | runtime.goexit
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +Wraps: (2) directory to store dumps could not be determined
E210802 18:32:58.622098 417449 server/server_sql.go:306  [n?] 34 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E210802 18:33:05.337459 461455 jobs/adopt.go:327  [n1] 35  job 681111666415370241: adoption completed with error could not remove enum value "us-east3" as it is being used in the partitioning of index rbr@primary
    region_test.go:734: condition failed to evaluate within 45s: unexpected number of regions, expected: [us-east1 us-east2] found [us-east1 us-east2 us-east3]
        goroutine 417449 [running]:
        runtime/debug.Stack(0xc0135e7cb8, 0x663ba40, 0xc019ad5dd0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x66f9df8, 0xc00be0fc80, 0xc0135e7cb8)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1(0xc00be0fc80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:734 +0x9fa
        testing.tRunner(0xc00be0fc80, 0xc01546b230)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
E210802 18:33:50.994311 439400 jobs/registry.go:644  [-] 36  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210802 18:33:50.995940 443076 jobs/registry.go:657  [-] 37  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210802 18:33:50.996366 439400 jobs/registry.go:657  [-] 38  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210802 18:33:50.996940 439398 jobs/registry.go:624  [-] 39  error expiring job sessions: expire-sessions: context canceled
E210802 18:33:50.997035 439398 jobs/registry.go:632  [-] 40  failed to serve pause and cancel requests: context canceled
E210802 18:33:50.997823 442000 jobs/registry.go:624  [-] 41  error expiring job sessions: expire-sessions: context canceled
E210802 18:33:50.997934 442000 jobs/registry.go:632  [-] 42  failed to serve pause and cancel requests: context canceled
E210802 18:33:51.000582 441230 jobs/registry.go:644  [-] 43  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210802 18:33:51.000843 441230 jobs/registry.go:657  [-] 44  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210802 18:33:51.011280 418196 jobs/registry.go:624  [-] 46  error expiring job sessions: expire-sessions: context canceled
E210802 18:33:51.011652 418196 jobs/registry.go:632  [-] 47  failed to serve pause and cancel requests: context canceled
E210802 18:33:51.011167 420832 jobs/registry.go:632  [-] 45  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210802 18:33:51.016449 419961 jobs/registry.go:657  [-] 48  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210802 18:33:51.017852 420834 jobs/registry.go:657  [-] 49  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210802 18:33:51.018637 419799 kv/kvserver/reports/reporter.go:143  [n3,replication-reporter] 50  failed to generate replication reports: failed to save range status report: upsert-replication-stats: collecting referenced descriptors: failed to send RPC: sending to all replicas failed; last error: failed to connect to n1 at 127.0.0.1:40085: context canceled
    --- FAIL: TestRegionAddDropWithConcurrentBackupOps/drop-region-succeed-backup-database (61.28s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRegionAddDropWithConcurrentBackupOps PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/multiregion ajstorm
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropWithConcurrentBackupOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropWithConcurrentBackupOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254273&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254273&tab=artifacts#/) on master @ [1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2](https://github.com/cockroachdb/cockroach/commits/1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2):


```
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/tracedumper/tracedumper.go:116
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/server.newSQLServer
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server_sql.go:304
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/server.NewServer
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:711
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:1320
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:289
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:457
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:271
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:174
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils.TestingCreateMultiRegionCluster
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go:78
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:711
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | testing.tRunner
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/usr/local/go/src/testing/testing.go:1193
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | runtime.goexit
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +Wraps: (2) directory to store dumps could not be determined
E210803 01:28:55.303643 406603 server/server_sql.go:306  [n?] 29 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E210803 01:29:03.030140 455922 jobs/adopt.go:327  [n1] 30  job 681193447195443201: adoption completed with error could not remove enum value "us-east3" as it is being used in the partitioning of index rbr@primary
    region_test.go:734: condition failed to evaluate within 45s: unexpected number of regions, expected: [us-east1 us-east2] found [us-east3 us-east1 us-east2]
        goroutine 406603 [running]:
        runtime/debug.Stack(0xc00f187cb8, 0x6640960, 0xc00d580690)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x66fedf8, 0xc00c4a7680, 0xc00f187cb8)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1(0xc00c4a7680)
        	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:734 +0x9fa
        testing.tRunner(0xc00c4a7680, 0xc00c1ff788)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
E210803 01:29:48.582676 432498 jobs/registry.go:644  [-] 31  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210803 01:29:48.583933 432498 jobs/registry.go:657  [-] 32  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210803 01:29:48.585453 433508 jobs/registry.go:624  [-] 33  error expiring job sessions: expire-sessions: context canceled
E210803 01:29:48.585556 433508 jobs/registry.go:632  [-] 34  failed to serve pause and cancel requests: context canceled
E210803 01:29:48.590901 434577 jobs/registry.go:657  [-] 35  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210803 01:29:48.593513 433510 jobs/registry.go:644  [-] 36  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210803 01:29:48.593792 433510 jobs/registry.go:657  [-] 37  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210803 01:29:48.598752 434575 jobs/registry.go:632  [-] 38  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210803 01:29:48.622565 409019 jobs/registry.go:632  [-] 39  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210803 01:29:48.623060 408535 jobs/registry.go:624  [-] 40  error expiring job sessions: expire-sessions: context canceled
E210803 01:29:48.623379 408535 jobs/registry.go:632  [-] 41  failed to serve pause and cancel requests: context canceled
    --- FAIL: TestRegionAddDropWithConcurrentBackupOps/drop-region-succeed-backup-database (63.12s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRegionAddDropWithConcurrentBackupOps PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/multiregion ajstorm
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropWithConcurrentBackupOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropWithConcurrentBackupOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254637&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254637&tab=artifacts#/) on master @ [1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2](https://github.com/cockroachdb/cockroach/commits/1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2):


```
E210803 05:24:35.993972 358138 server/server_sql.go:306  [n?] 24 +Wraps: (2) directory to store dumps could not be determined
E210803 05:24:35.993972 358138 server/server_sql.go:306  [n?] 24 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25  failed to initialize trace dumper: directory to store dumps could not be determined
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +(1) attached stack trace
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  -- stack trace:
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/server/tracedumper.NewTraceDumper
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/tracedumper/tracedumper.go:116
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/server.newSQLServer
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server_sql.go:304
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/server.NewServer
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:711
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:1320
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:289
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:457
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:271
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:174
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils.TestingCreateMultiRegionCluster
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go:78
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:711
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | testing.tRunner
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/usr/local/go/src/testing/testing.go:1193
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | runtime.goexit
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +Wraps: (2) directory to store dumps could not be determined
E210803 05:24:36.075048 358138 server/server_sql.go:306  [n?] 25 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E210803 05:24:43.954407 399722 jobs/adopt.go:327  [n1] 26  job 681239784851996673: adoption completed with error could not remove enum value "us-east3" as it is being used in the partitioning of index rbr@primary
    region_test.go:734: condition failed to evaluate within 45s: unexpected number of regions, expected: [us-east1 us-east2] found [us-east1 us-east2 us-east3]
        goroutine 358138 [running]:
        runtime/debug.Stack(0xc004889cb8, 0x6640960, 0xc019959c38)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x66fedf8, 0xc00c9c9980, 0xc004889cb8)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionAddDropWithConcurrentBackupOps.func1(0xc00c9c9980)
        	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/region_test.go:734 +0x9fa
        testing.tRunner(0xc00c9c9980, 0xc0014f9278)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
E210803 05:25:29.386238 377033 jobs/registry.go:624  [-] 27  error expiring job sessions: expire-sessions: context canceled
E210803 05:25:29.386366 377033 jobs/registry.go:632  [-] 28  failed to serve pause and cancel requests: context canceled
E210803 05:25:29.403089 377716 jobs/registry.go:632  [-] 29  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210803 05:25:29.411723 361106 jobs/registry.go:644  [-] 30  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210803 05:25:29.411935 361106 jobs/registry.go:657  [-] 31  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    --- FAIL: TestRegionAddDropWithConcurrentBackupOps/drop-region-succeed-backup-database (59.50s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRegionAddDropWithConcurrentBackupOps PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/multiregion ajstorm
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropWithConcurrentBackupOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/multiregionccl/region_test.go
+++ b/pkg/ccl/multiregionccl/region_test.go
@@ -860,7 +860,6 @@ CREATE DATABASE db WITH PRIMARY REGION "us-east1" REGIONS "us-east2" PLACEMENT R
 // - Validate that the database and its tables look as expected.
 func TestRegionAddDropWithConcurrentBackupOps(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 68341, "flaky test")
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t, "times out under race")
@@ -982,7 +981,7 @@ INSERT INTO db.rbr VALUES (1,1),(2,2),(3,3);
 							defer mu.Unlock()
 							if !regionAlterCmd.shouldSucceed {
 								// Trigger a roll-back.
-								return errors.New("nope")
+								return jobs.MarkAsPermanentJobError(errors.New("nope"))
 							}
 							// Trod on.
 							return nil
@@ -1020,7 +1019,9 @@ INSERT INTO db.rbr VALUES (1,1),(2,2),(3,3);
 					dbRegions := make([]string, 0, len(regionAlterCmd.expectedPartitions))
 					rowsRegions, err := sqlDBRestore.Query("SELECT region FROM [SHOW REGIONS FROM DATABASE db]")
 					require.NoError(t, err)
-					defer rowsRegions.Close()
+					defer func() {
+						require.NoError(t, rowsRegions.Close())
+					}()
 					for {
 						done := rowsRegions.Next()
 						if !done {

